File: internal/sem/testdata/langmatrix/rust/ledger.rs
pub struct Ledger {
    pub total: i32,
}

impl Ledger {
    pub fn add(&self, amount: i32) -> i32 {
        self.total + amount
    }
}

pub fn ledger_double(amount: i32) -> i32 {
    let ledger = Ledger { total: 0 };
    ledger.add(amount) * 2
}

File: internal/sem/testdata/langmatrix/zig/ledger.zig
const std = @import("std");

pub const Ledger = struct {
    total: i32,

    pub fn add(self: Ledger, amount: i32) i32 {
        return self.total + amount;
    }
};

pub fn ledgerDouble(amount: i32) i32 {
    const ledger = Ledger{ .total = 0 };
    return ledger.add(amount) * 2;
}
